src/types/network_type.rs

```diff
@@ -57,3 +57,27 @@ impl fmt::Display for NetworkType {
         write!(f, "{}", self.to_str())
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct NetworkInfo {
+    pub network_type: NetworkType,
+    pub url: String,
+}
+
+impl NetworkInfo {
+    pub fn new(network_type: NetworkType, url: String) -> Self {
+        Self { network_type, url }
+    }
+    pub fn mainnet() -> Self {
```

Comment:
please add fn doc and notice user to use a self hosted ckb node in production, `ckb.dev` is for testing purposes only.